### PR TITLE
Check for traversable data

### PR DESCRIPTION
**Pull request by ghost**
With this change, phalcon framework become easy to use this lib

### CODE REVIEW

**Comment by neomerx** [comment]
@sumeko Can you please add a simple test for converting something which is not an array but Traversable just to make sure it won't fail in other places?

**Comment by neomerx** [comment]
@sumeko I guess [one of those](https://github.com/neomerx/json-api/blob/master/tests/Encoder/EncoderTest.php) might be used as a basis

File: src/Encoder/Parser/Parser.php [hunk 1]
```diff
@@ -131,7 +131,7 @@ private function parseData($data)
         } elseif ($this->isRefFrame()) {
             yield $this->createReplyForRef();
         } else {
-            if (is_array($data) === true) {
+            if ((is_array($data) === true) or ($data instanceof \Traversable)) {
```
Comment (neomerx) [inline src/Encoder/Parser/Parser.php:134]: please use `||` instead of `or`
please import \Traversable with `use` directive

**Comment by neomerx** [comment]
Out of curiosity I've added the test. It seems it will require more than this change. Why Falcon can't work with arrays?

``` php

    /**
     * Test encode Traversable collection of resource(s).
     */
    public function testEncodeTraversableObjectsWithAttributesOnly()
    {
        $author   = Author::instance(9, 'Dan', 'Gebhardt');
        $endcoder = Encoder::instance([
            Author::class => function ($factory, $container) {
                $schema = new AuthorSchema($factory, $container);
                $schema->linkRemove(Author::LINK_COMMENTS);
                return $schema;
            }
        ], $this->encoderOptions);

        $itemSet = new ArrayIterator([$author]);
        $actual  = $endcoder->encode($itemSet);

        $expected = <<<EOL
        {
            "data" : [
                {
                    "type"       : "people",
                    "id"         : "9",
                    "attributes" : {
                        "first_name" : "Dan",
                        "last_name"  : "Gebhardt"
                    },
                    "links" : {
                        "self" : "http://example.com/people/9"
                    }
                }
            ]
        }
EOL;
        // remove formatting from 'expected'
        $expected = json_encode(json_decode($expected));

        $this->assertEquals($expected, $actual);
    }
```

**Comment by nueko** [comment]
Phalcon Model returning \Phalcon\Mvc\Model\Resultset\Simple that is not an array, when fetching rows. so checking with is_array function it returns false

**Comment by neomerx** [comment]
@sumeko I've added a new branch `feature/issue36` which I hope should work fine with phalcon data. Can you please check it? Thanks

**Comment by nueko** [comment]
quick checking.. great, that's work!

**Comment by nueko** [comment]
here if you want to checkit out [phalcon-jsonapi](https://github.com/sumeko/phalcon-jsonapi)

just run the PHP built-in 

``` bash
php -S 127.0.0.1:8000 -t app/public
```

**Comment by neomerx** [comment]
I'll add more tests and then merge

**Comment by neomerx** [comment]
I'd like to try. Can you please add composer support rather than git-install?

**Comment by neomerx** [comment]
@sumeko merged to master and released as v0.5.3

**Pull request closed without merging**